### DISCUSSION TITLE
"GitHub Copilot requires VS Code 1.57+" when using VS Code 1.61.0

### DISCUSSION THREAD

**Discussion by Pablo1107** [community/community#6684] [Copilot Conversations] [answered]
Yesterday I tried to install and use GitHub Copilot. First I tried to install it from VS Code Marketplace and it says "No extensios found", I install it through VSIX. And after authorize the GitHub login I get this message:

`GitHub Copilot requires VS Code 1.57+. Please update your VS Code`

![image](https://user-images.githubusercontent.com/17091659/138604129-6a35c988-2ca0-4c59-ae61-16344840a5fd.png)

But I have version 1.61.0...

![image](https://user-images.githubusercontent.com/17091659/138604176-bbcc2570-93ac-4299-b762-64b0d136e71e.png)

**Comment by Krzysztof-Cieslak** [comment, marked as answer]
Please check this answer - https://github.com/github/feedback/discussions/6629#discussioncomment-1524627

**Comment by omidshojaee** [comment]
Hello,

I have this issue right now and proposed solution is not working for me.

![Screenshot 2022-04-22 152551](https://user-images.githubusercontent.com/31741561/164701671-68ecc6ba-1067-4c15-9303-74138f2346c2.png)

![Screenshot 2022-04-22 152449](https://user-images.githubusercontent.com/31741561/164701505-f5be3e87-0b6a-4280-abd2-fdcf5d28d548.png)

![Screenshot 2022-04-22 152405](https://user-images.githubusercontent.com/31741561/164701540-6a0e2ec3-f997-42c6-8ffe-805ed0752ba0.png)

**Comment by GWDx** [comment]
> OS: Ubuntu 20.04
> VSCodium

I also meet the same problem. `GitHub Copilot requires VS Code 1.57+`. I solve it.



It is noticed that in the developer window, a critical warning mentions `product.json`

```
CRITI Extension 'github.copilot' appears in product.json but enables LESS API proposals than the extension wants.
package.json (LOSES): inlineCompletionsNew, inlineCompletionsAdditions
product.json (WINS): inlineCompletions, textDocumentNotebook
```

**Solution**

Update `/usr/share/codium/resources/app/product.json`, add the lost attributes.

```
"GitHub.copilot": ["inlineCompletions", "inlineCompletionsNew", "inlineCompletionsAdditions", "textDocumentNotebook"],
```

**Reply by onepisYa** [reply]
> me too
> OS: MacOS 12.3.1 (21E258)

 code-insiders -v
1.67.0-insider
8b7086afdbfd42bf6e7b8c0bb586530bf7cf1396
arm64

any good suggestions for me

**Reply by BettyJJ** [reply]
I got this:

>   ERR Extension 'github.copilot' wants API proposal 'inlineCompletionsNew' but that proposal DOES NOT EXIST. Likely, the proposal has been finalized (check 'vscode.d.ts') or was abandoned.

Returning to an older version of VSCode worked for me.
